### ISSUE TITLE
Nicer infix operators?

### ISSUE THREAD

**Issue by kindaro** [feuerbach/tasty#262]
I know there is a strong sentiment in the Haskell community towards legacy support, backwards compatibility and so on, and I am prepared to discover that this is a _"wontfix"_, but I would like to express my vote anyway. 

1. HUnit has operators `@?=` and `@=?`. I propose we change them to `?=` and `=?` — it would be so much more readable and intuitive. _(Which side the question, the same side the value being checked.)_ Hoogle does report a couple of packages using these names, but I claim that a test engine is used far more widely _(ideally, in every package)_ and so it is fair to have the shorter names.

2. Let us redefine `QuickCheck.===` to mean extrinsic equality for functions: `f === g = \x -> f x == g x`. Properties are quite often expressed as equality of functions anyway — think of the laws of any standard class.

**Comment by feuerbach** [comment]
Changing the existing module is indeed not an option.

However, I am all for a redesign (under a new name — say, Test.Tasty.HUnit2) with better names etc.

There are also some functions/operators that I miss:

* shouldSatisfy from HSpec
* something like `assertLeft :: Either a b -> (a -> Assertion) -> Assertion` and `assertRight :: Either a b -> (b -> Assertion) -> Assertion`
* go through hunit and hspec and see if there's anything else worth borrowing

So if you feel like having a go at redesigning the HUnit module with the above considerations in mind, please do.

**Comment by kindaro** [comment]
So if such a patch were to appear, there is a reasonable expectation that it will be merge? If such is the case, then I am putting it in my to do list.

**Comment by feuerbach** [comment]
Yes, I think we've been stuck with the original HUnit design for too long, so a well designed replacement would be welcome.

**Comment by fendor** [comment]
I would like to second more assertion primitives, e.g. appreciate something like `hspec`'s `shouldReturn`, similarly `shouldThrow`, etc...

Additionally, in my opinion real names are preferable to infix operators. Especially when others such as `shouldReturn`, `shouldSatisfy` and `shouldThrow` are added which maybe have no nice infix operator.
So, I would at least would like to have synonyms for infix operators such as `=?` and `?=`.

**Comment by re-xyr** [comment]
Is this still open to PRs? I'd like to implement `hspec`-style predicates for `tasty-hunit`.

**Comment by sol** [comment]
`hspec-expectations` builds on top of `HUnit` and it does not depend on `hspec-core` or any other parts of `hspec`.

This is by design so that `hspec-expectations` can be used with any testing library that uses `HUnit`.

Instead of reimplementing `hspec-expectations` in Tasty, please consider to revisit #327.

**Comment by re-xyr** [comment]
Yes, that was what I initially thought... the divide between `tasty-hunit` and `hunit` is really confusing. But then I read that this change was intentional (#230). Some years have passed since though, and it's possble that @UnkindPartition will change their mind.
